x-pack/plugins/security/public/nav_control/nav_control_service.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Subscription, Observable } from 'rxjs';
+import { CoreStart } from 'src/core/public';
+import ReactDOM from 'react-dom';
+import React from 'react';
+import { LicensingPluginSetup, ILicense } from '../../../licensing/server';
+import { SecurityLicenseService } from '../../common/licensing';
+import { AuthenticatedUser } from '../../common/model';
+import { SecurityNavControl } from './nav_control_component';
+
+interface SetupDeps {
+  securityLicenseService: ReturnType<SecurityLicenseService['setup']>;
+  licensing: LicensingPluginSetup;
+}
+
+interface StartDeps {
+  core: CoreStart;
+}
+
+export class SecurityNavControlService {
+  private securityLicenseService!: ReturnType<SecurityLicenseService['setup']>;
+
+  private license$!: Observable<ILicense>;
```

Comment:
How do you feel about changing this to only need an instance of the `SecurityLicenseService` per Aleh's comment here: https://github.com/elastic/kibana/pull/52386#discussion_r355929572 ?

Reply:
++ I'm on board!